scm-ui/ui-components/src/markdown/MarkdownView.tsx

```diff
@@ -155,20 +164,148 @@ class MarkdownView extends React.Component<Props, State> {
       rendererList.code = MarkdownCodeRenderer;
     }
 
-    const plugins = [...mdastPlugins, createTransformer(t)].map(createMdastPlugin);
+    // NEW IMPL START
 
-    const baseProps = {
-      className: "content is-word-break",
-      renderers: rendererList,
-      plugins: [gfm],
-      astPlugins: plugins,
-      children: content
-    };
+    const components: { [key: string]: any } = {};
+
+    if (rendererList.code) {
+      const codeRendererAdapter = function({ node, children }: any) {
```

Comment:
Could we extract all this new factories to a new file like `markdownRenderers.ts`?